### PR TITLE
lightning: use SprintfWithIdentifier to process strings with identifier

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #30699

Problem Summary:

### What changed and how does it work?

This PR also remove glue_checkpoint.go because its feature is replaced by IMPORT INTO

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by lyzx2001 — APPROVED** [review]

File: br/pkg/lightning/checkpoints/checkpoints.go [hunk 1]
```diff
@@ -1601,26 +1596,24 @@ func (cpdb *MySQLCheckpointsDB) GetLocalStoringTables(ctx context.Context) (map[
 
 // IgnoreErrorCheckpoint implements CheckpointsDB.IgnoreErrorCheckpoint.
 func (cpdb *MySQLCheckpointsDB) IgnoreErrorCheckpoint(ctx context.Context, tableName string) error {
-	var colName string
+	var query, query2 string
 	if tableName == allTables {
-		// This will expand to `WHERE 'all' = 'all'` and effectively allowing
-		// all tables to be included.
-		colName = stringLitAll
+		query = common.SprintfWithIdentifiers("UPDATE %s.%s SET status = ? WHERE 'all' = ? AND status <= ?", cpdb.schema, CheckpointTableNameEngine)
```
Comment (D3Hunter) [inline br/pkg/lightning/checkpoints/checkpoints.go:1601]: column name cannot be used as bind var

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/checkpoints/checkpoints.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/lightning/checkpoints/checkpoints.go:1601]: here `?` is a string, not column name. For `--checkpoint-error-destroy=all` the expected SQL is `WHERE 'all' = 'all'`. For ``--checkpoint-error-destroy='`schema`.`table`'`` the expected SQL is ``WHERE table_name = '`schema`.`table`'``

Comment (D3Hunter) [reply br/pkg/lightning/checkpoints/checkpoints.go:1601]: > the expected SQL is WHERE 'all' = 'all'.

why not remove it?

Comment (lance6716) [reply br/pkg/lightning/checkpoints/checkpoints.go:1601]: below `ExecContext(c, query, CheckpointStatusLoaded, tableName, CheckpointStatusMaxInvalid)` can use same arguments for the two types of query.

Comment (D3Hunter) [reply br/pkg/lightning/checkpoints/checkpoints.go:1601]: create a `args1`/`args2` ?

Comment (lance6716) [reply br/pkg/lightning/checkpoints/checkpoints.go]: https://github.com/pingcap/tidb/pull/50434/commits/42bdb4dc7731f4296c6adb033a169adbaefedb54

**Comment by lance6716** [comment]
ping @D3Hunter

**Pull request merged**